lib/galaxy/objectstore/__init__.py

```diff
@@ -712,54 +708,24 @@ def parse_xml(clazz, config_xml, legacy=False):
             'backends': backends,
         }
 
-        for elem in [e for e in backends_root if e.tag == 'backend']:
-            id = elem.get('id')
-            weight = int(elem.get('weight', 1))
-            maxpctfull = float(elem.get('maxpctfull', 0))
-            elem_type = elem.get('type', 'disk')
-            store_by = elem.get('store_by', None)
-            if elem_type:
-                path = None
-                extra_dirs = []
-                for sub in elem:
-                    if sub.tag == 'files_dir':
-                        path = sub.get('path')
-                    elif sub.tag == 'extra_dir':
-                        type = sub.get('type')
-                        extra_dirs.append({"type": type, "path": sub.get('path')})
-
-                backend_dict = {
-                    'id': id,
-                    'weight': weight,
-                    'max_percent_full': maxpctfull,
-                    'files_dir': path,
-                    'extra_dirs': extra_dirs,
-                    'type': elem_type,
-                }
-                if store_by is not None:
-                    backend_dict['store_by'] = store_by
-                backends.append(backend_dict)
+        for b in config_xml.find('backends'):
```

Comment:
Is that correct also if legacy is true ? Should this be `config_xml.findall(backends)` or `backends_root.find('backends') ` maybe ? In your PR it looks like `backends_root ` is now unused, in which case you can remove this.

Reply:
backends_root is used I setting a field of config_dict. I'll double check the legacy true/false logic again to see if its correct.

Reply:
Ok, I looked into this a bit more. I had removed the from_xml(), but upon further inspection, I think that may not be right. In from_xml(), if config_xml is None, we read off a config file specified by distributed_object_store_file field of config (and we set the legacy fag to True). For backward compatibility, I added the method back (Unless you guys are sure its a stale code and can be removed with no side effect). 

I also reverted the for loop's first line (the one that iterates over backends). 

This should make my changes fully backward compatible.

Reply:
Awesome, yes, looks good.